src/bot.ts

```diff
@@ -141,9 +142,14 @@ export function createBot(config: BotConfig) {
       return path.resolve(config.dataDir, `${content}.mp3`);
     }
     if (config.myInstantsEnabled && content.startsWith("instant")) {
-      const sound = encodeURIComponent(content.substr("instant".length).trim().substr(0, 256))
+      const sound = encodeURIComponent(content.slice("instant".length).trim().slice(0, 256))
       return `https://www.myinstants.com/media/sounds/${sound}.mp3`;
     }
+    if (config.wikiaCDNEnabled && content.startsWith("wikiacdn")) {
+      const link = content.slice("wikiacdn".length).trim().slice(0, 256);
+      if (link.startsWith("https://static.wikia.nocookie.net/"))
+        return link;
```

Comment:
I'm confused. Does this feature simply play the url that is passed, after checking whether it belongs to `static.wikia.nocookie`? Is there a chance to make this more accessible, similar to `instant fart`? In instant buttons case we take a url part and build the rest on the bot side.

Reply:
> I'm confused. Does this feature simply play the url that is passed, after checking whether it belongs to `static.wikia.nocookie`? Is there a chance to make this more accessible, similar to `instant fart`? In instant buttons case we take a url part and build the rest on the bot side.

Without parsing a web page, not really. the rest of the urls usually have a couple of subfolders with no obvious logic. we can ask for a specific subdomain of wikia/fandom.com but it we would still require a good chunk of url from the user.
Some sample links, 
[https&#58;//static.wikia.nocookie.net/ageofempires/images/0/0f/024_are_we_there_yet.mp3/revision/latest?cb=20210202072742](https&#58;//static.wikia.nocookie.net/ageofempires/images/0/0f/024_are_we_there_yet.mp3/revision/latest?cb=20210202072742)
[https&#58;//static.wikia.nocookie.net/ageofempires/images/1/16/028_throw_things_at_me.mp3/revision/latest?cb=20210202072812](https&#58;//static.wikia.nocookie.net/ageofempires/images/1/16/028_throw_things_at_me.mp3/revision/latest?cb=20210202072812)

Reply:
On another note, I think it might be good to add an option that allows for any audio file link. If the access to the bot commands are restricted properly to the trustworthy channel members it might be smart to allow it.

Reply:
There's actually a better way to find the file URL without requiring any HTML parsing. Let's use the [Mediawiki API](https://en.wikipedia.org/w/api.php?action=help&modules=query%2Bimageinfo) for this, e.g.:

https://ageofempires.fandom.com/api.php?action=query&titles=File:024_are_we_there_yet.mp3&prop=imageinfo&iiprop=url|size|mediatype -->It returns all items matching the title across any page under the ageofempires sub-domain. 

So I'd suggest changing this prompt to take only one parameter which is the file name (without the "File:" part), e.g., invoked such as `aoefandom 024_are_we_there_yet.mp3`.  Then call the api.php to extract the `static.wiki.nocookie.net` url.

Before attempting to play the media URL, let's also add some validations and only attempt to play the sound if;
- "size" attribute is smaller than maybe a few hundred KBs at most
- "mediatype" attribute is "AUDIO"

Reply:
About allowing open links, let's not have it for now... There's no way to tell who's "trustworthy" right now, and I don't want leave a bigger gap to play rather obnoxious stuff in our channel :)